src/daft-sql/src/functions.rs

```diff
@@ -274,9 +280,18 @@ impl<'a> SQLPlanner<'a> {
                 unsupported_sql_err!("subquery function argument")
             }
             sqlparser::ast::FunctionArguments::List(args) => {
-                if args.duplicate_treatment.is_some() {
-                    unsupported_sql_err!("function argument with duplicate treatment");
+                let fn_is_count = fn_name.as_str() == "count";
+                let using_duplicate_treatment =
+                    matches!(args.duplicate_treatment, Some(DuplicateTreatment::Distinct));
+
+                match (fn_is_count, using_duplicate_treatment) {
+                    (true, true) => {
+                        fn_match = get_func_from_sqlfunctions_registry("count_distinct")?;
+                    }
+                    (false, true) => unsupported_sql_err!("DISTINCT is only supported on COUNT"),
+                    (true, false) | (false, false) => (),
```

Comment:
Just so that we also check if someone passes in something like `max(ALL x)`, could you match something like this instead:
```suggestion
                match (fn_name.as_str(), args.duplicate_treatment) {
                    ("count", Some(DuplicateTreatment::Distinct)) => {
                        fn_match = get_func_from_sqlfunctions_registry("count_distinct")?;
                    }
                    ("count", _) => {},
                    (_, Some(_)) => unsupported_sql_err!("DISTINCT is only supported on COUNT"),
                    _ => {}
```

Reply:
Updated.